unitex/src/fr/umlv/unitex/tfst/tagging/TaggingModel.java

```diff
@@ -105,27 +150,677 @@ protected void updateModel() {
+	 */		
+	private int findNewBegin(ArrayList<Context> lstContext) {
+		boolean hasSpace = false;
+		
+		for (Context context : lstContext) {
+				if(context.space)
+					hasSpace = true;
+				
+				context.space = false;
+		}
+		
+		if(hasSpace)
+			return 2;
+		
+		for (Context context : lstContext) {
+			if(context.pos == 0) {
+				if(boxes[sortedNodes[context.path.get(context.currentBox - 1)]].getBounds() != null)
+					return boxes[sortedNodes[context.path.get(context.currentBox)]].getBounds().getStart_in_tokens() -
+							boxes[sortedNodes[context.path.get(context.currentBox - 1)]].getBounds().getEnd_in_tokens();
+				else
+					return boxes[sortedNodes[context.path.get(context.currentBox)]].getBounds().getStart_in_tokens();
+		
+			}
+		}
+		return -1;
+	}
+
+	private boolean verifyBox(GenericGraphBox box, ArrayList<Context> copyContext) {
+		String txt = Normalizer.normalize(getTextBoxe(box), Normalizer.Form.NFD);
+		int size = txt.length();
+		ArrayList<Context> ctxtRemove = new ArrayList<>();
+		if(txt.equals("<E>"))
+			return true;
+		
+		for (Context context : copyContext) {
+			
+			if(context.currentBox >= context.path.size()) {
+				ctxtRemove.add(context);
+				continue;
+			}
+			int numBox = context.path.get(context.currentBox);
+			String txtBis = Normalizer.normalize(getTextBoxe(boxes[sortedNodes[numBox]]), Normalizer.Form.NFD);
+				
+			for(int i = 0; i < size; i++) {
+				if(context.currentBox >= context.path.size()) {
+					ctxtRemove.add(context);
+					break;
+				}
+				numBox = context.path.get(context.currentBox);
+				txtBis =  Normalizer.normalize(getTextBoxe(boxes[sortedNodes[numBox]]), Normalizer.Form.NFD);
+
+				if(context.space == true && i != 0) {
+					if(txt.charAt(i) != ' ') 
+						ctxtRemove.add(context);
+					context.space = false;
+				}
+				else {
+					if(context.pos >= txtBis.length()) {
+						context.pos = 0;
+						context.currentBox ++;
+					}
+					if(txt.charAt(i) != txtBis.charAt(context.pos)) {
+						ctxtRemove.add(context);
+					}
+					context.pos ++;
+					if(context.pos >= txtBis.length()) {
+						if(context.currentBox + 1 < context.path.size() && boxes[sortedNodes[context.path.get(context.currentBox+1)]].type != GenericGraphBox.FINAL
+								&& boxes[sortedNodes[context.path.get(context.currentBox+1)]].getBounds().getStart_in_tokens() -
+								boxes[sortedNodes[numBox]].getBounds().getEnd_in_tokens() == 2)
+							context.space = true;
+						context.pos = 0;
+						context.currentBox ++;
+					}
+				}
+			}
+			if(context.pos < txtBis.length() && txtBis.charAt(context.pos) == ' ') {
+				context.space = true;
+				context.pos++;
+			}
+		}
+		
+		if(taggingStates[box.getBoxNumber()] != TaggingState.TO_CHECK)
+			return true;
+		
+		for (Context context : ctxtRemove)
+			copyContext.remove(context);
+
+		if(copyContext.isEmpty())
+			return false;
+		
+		return true;
+	}
+	
+	
+	/** @Aissa
+	 * 
+	 * This function is used to find all sentences between the 2 factorizations box 
```

Comment:
sentence --> sequence, factorization(s) box --> factorization node, the number of boxes of --> the number of each box in